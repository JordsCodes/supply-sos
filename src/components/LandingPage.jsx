import React from "react";
import FadeIn from "react-fade-in";
import "../styles/prompt.css";

function LandingPage() {
  return (
    <FadeIn>
      <div className="prompt">
        <h2 className="prompt_heading">No planning? No problem.</h2>
        <p className="prompt_content">
          Downloadable, customisable AI-generated lesson plans for Key Stages 1
          and 2.
        </p>
      </div>
    </FadeIn>
  );
}

export default LandingPage;
