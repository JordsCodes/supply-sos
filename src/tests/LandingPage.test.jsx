import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LandingPage from "../components/LandingPage";

describe("Landing Page", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<MemoryRouter><LandingPage /></MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("heading renders correctly", () => {
    render(<MemoryRouter><LandingPage /></MemoryRouter>
    );
    const heading = screen.getByText("No planning? No problem.");

    expect(heading).toBeInTheDocument();
  });

  test("paragraph renders correctly", () => {
    render(<MemoryRouter><LandingPage /></MemoryRouter>
    );
    const paragraph = screen.getByText(
      "Downloadable, customisable AI-generated lesson plans for Key Stages 1 and 2.",
    );

    expect(paragraph).toBeInTheDocument();
  });

  test("button renders correctly", () => {
    render(<MemoryRouter><LandingPage /></MemoryRouter>
    );
    const button = screen.getByText("Let's get started");

    expect(button).toBeInTheDocument();
  });

  test("image renders correctly", () => {
    render(<MemoryRouter><LandingPage /></MemoryRouter>
    );
    const image = screen.getByRole("img");

    expect(image).toBeInTheDocument();
  });
});
