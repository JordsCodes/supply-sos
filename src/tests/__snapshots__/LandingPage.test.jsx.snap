// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="landing-page"
  >
    <div
      class="welcome"
    >
      <h2
        class="welcome_heading"
      >
        No planning? No problem.
      </h2>
      <p
        class="welcome_text"
      >
        Downloadable, customisable AI-generated lesson plans for Key Stages 1 and 2.
      </p>
      <a
        class="button"
        href="/year-group"
      >
        Let's get started
      </a>
    </div>
    <img
      alt="teacher in classroom"
      class="image"
      src="landing-page.png"
    />
  </div>
</DocumentFragment>
`;
